### ISSUE TITLE
ButtonMethod attribute doesn't work with Mirror's NetworkBehaviour

### ISSUE THREAD

**Issue by CorneliusCornbread** [Deadcows/MyBox#114]
So I've spent the past couple of hours completely confused as to why `[ButtonMethod]` doesn't show up in the inspector inside of classes that inherit from [Mirror](https://github.com/vis2k/Mirror)'s [NetworkBehaviour](https://github.com/vis2k/Mirror/blob/master/Assets/Mirror/Runtime/NetworkBehaviour.cs). I tested inheriting from abstract MonoBehaviour classes, I've tested inheriting MonoBehaviour type classes from the Mirror namespace. All of these worked, but for some reason inheriting specifically from NetworkBehaviour caused the buttons to not appear with `[ButtonMethod]`.

However, `[DisplayInspector]` and `[ButtonMethod]`s inside those displayed inspectors work perfectly fine. Changing the class to a MonoBehaviour solves the issues with buttons not showing up. I genuinely have no clue as to why this could be happening, it's only classes that specifically inherit from NetworkBehaviour that causes this. Is it possible a custom drawer for NetworkBehaviour is messing with `[ButtonMethod]`?

TL;DR classes that inherit from NetworkBehaviour and try to use `[ButtonMethod]` won't have the button show in the inspector. However, you can make buttons appear in serialized classes with `[DisplayInspector]`

My Mirror version is 30.2.1, the latest on the asset store as of 12/25/2020. 
My box version 1.5.0, Unity package, latest as far as I can tell.

**Comment by Deadcows** [comment]
ButtonMethod is handled via CustomEditor for all objects of Object type (which is all MonoBehaviours and ScriptableObjects). And unity can use only one custom editor per type... so yeah, his "NetworkBehaviourInspector" custom editor is in priority and mine is discarded. 
So I'd say it's not possible.. however!
If you really wanna make this work, you may create your own type on top of the NetworkBehaviour and create custom drawer for it. ButtonMethodHandler for the rescue, will make it pretty easy to do! I just typing it out of my head so this code might be not compillable, but to get the idea:

```

public abstract class MyNetworkBehaviour : NetworkBehaviour {}

[CustomEditor(typeof(MyNetworkBehaviour), true), CanEditMultipleObjects]
public class MyNetworkBehaviourInspector : Editor
{
        private ButtonMethodHandler _buttonMethod; 

        private void OnEnable()
        {
            _buttonMethod = new ButtonMethodHandler(target);
        }

        public override void OnInspectorGUI()
        {
            _buttonMethod?.OnBeforeInspectorGUI();            
            OnInspectorGUI();
            _buttonMethod?.OnAfterInspectorGUI();
        }
}

```

should do the trick. Now your MyNetworkBehaviourInspector is in priority and will use drawer of NetworkBehaviourInspector via OnInspectorGUI() method

**Comment by Deadcows** [comment]
Oh my, four months old request. Yeah, my answer is not on time, sorry for that 😑 But I'm on track now! Will answer faster next time 🙄
